src/validation/validator.ts

```diff
@@ -1 +1,24 @@
-export {};
+import { AnySchema } from 'yup/lib/schema'
+
+export class Validator {
+  private schema: AnySchema
+  private data: unknown
+  constructor(schema: AnySchema, data: unknown) {
+    this.schema = schema
+    this.data = data
+  }
+
+  isValid() {
+    return new Promise((resolve, reject) => {
+      this.schema
+        .validate(this.data)
+        .then(result => {
+          resolve(true)
+        })
+        .catch(error => {
+          console.error('error', error)
+          reject(false)
+        })
+    })
+  }
+}
```

Comment:
it is an overkill to wrap a small util in a class, 

we just need sth simple that takes the schema and data then returns the validation result!

check this example:

```js
import yup from 'yup'

export const validator = async ({
  schema,
  data
}: {
  schema: yup.ObjectSchema<any>
  data: any
}) => {
  const result = { isValid: true, error: '' }

  try {
    await schema.validate(data)
  } catch (error: unknown) {
    const validationError = error as yup.ValidationError

    result.isValid = false
    result.error = validationError.errors[0]
  }

  return result
}
```